plugins/custom-codecs/build.gradle

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+apply plugin: 'opensearch.opensearchplugin'
+apply plugin: 'opensearch.internal-cluster-test'
+
+opensearchplugin {
+  name 'custom-codecs'
```

Comment:
Should this be called as ```codecs-zstd``` similar to analysis plugins?

Reply:
Got it. Wondering if it would take away the flexibility of installing only the required codec the user wants. There are two different school of thoughts here.

1. Have ```common-codecs``` as a single plugin and have multiple codecs (```zstd```, ```snappy```, etc.) in it. Users can install this plugin and get the entire codec bundle.
2. Have separate plugin for each codec (for e.g.: ```codec-zstd```, ```codec-snappy```, etc.). Users can install the required ones, some user may want all and they can install all the plugins, some users may just want ```snappy``` or ```zstd``` and they will be able to install just that.

While no. 1 provides ease of installation and usage, no. 2 follows [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). Between them, I prefer no. 2 as it provides the user the ability to choose what they need (which was the trigger for this entire PR).

Reply:
Both points are valid, the one note regard point 1: if user does not specify the codec, the codec won't be loaded. The sad fact about most of the added codecs, they use JNI - so we have to allow plugin(s) to load basically any native library, having several plugins asking for that could be an issue. 

In favor of option 2, I think one of the options suggest by @nknize is to have `custom-codecs` repository where we could have dedicated plugins for each codec.

Reply:
> In favor of option 2, I think one of the options suggest by @nknize is to have custom-codecs repository where we could have dedicated plugins for each codec.

I like the idea, we should do that.